### ISSUE TITLE
question about for the task of anticipation on GTEA Gaze+  

### ISSUE THREAD

**Issue by Morethan123** [fpv-iplab/rulstm#8]
Hello.Your code is really a professional work and inspires me a lot. I am reproducing your work for the task of anticipation on GTEA Gaze+.There are some questions that i want to ask you.
  I notice that you choose to pick the 14 frame feature as your rulstm model input before the action happens in the dataset.py. i found here are time-stamps(frame of number you called) in training.csv. However,  in GTEA Gaze+, annotation files contains only verb label,noun label ,action label, and no time-stamps. I wonder know how to perform anticipation on GTEA Gaze+ and how to choose the extracted feature as the model's input.
  I guess there are difference between GTEA Gaze+ and EPIC-Kitchens.
  Hope to receive your reply.Thank you!

**Comment by antoninofurnari** [comment]
Hello,
Thank you for your interest in our work and for your kind words. We considered the "raw annotations" from EGTEA GAZE+ (https://www.dropbox.com/s/ksro6eqa6v59859/action_annotation.zip?dl=0&file_subpath=%2Fraw_annotations%2Faction_labels.csv). These include the starting and ending timestamps in milliseconds for each action segment.

Each action is associated to one verb and one or more nouns. To simplify, as with EPIC-KITCHENS, we only retailed the first noun. We also dumped each video to a fixed framerate of 30fps using ffmpeg and converted the timestamps to frame numbers.

Hope this helps,
Antonino

**Comment by Morethan123** [comment]
Hello,@antoninofurnari . You mention that dump each video to a fixed framerate of 30fps. To change the raw video to 30fps using ffmpeg since the raw video is 24fps or to extract directly the raw video to frames with 30fps as option using ffmpeg.  I am confused and wonder to know which one is your opinion you wish to express. 

I noticed that the action_labels.csv in "raw_annotations" file after your kind mention. It really provide the starting and ending time for each action clips. So i should train the TSN in the trimmed action clips for the task of action recognition in order to extract rgb_feature and flow feature. When finish this step, I should  follow the dataset.py to sample the sequnce length(14 in your paper) features before the action happens.  So, I should download the raw_video from EGTEA GAZE+ dataset to change the starting and ending timestamp in action_labels.csv into number of frames?

Hope to receive your reply! Thank you.

**Comment by antoninofurnari** [comment]
Yes, that's what I did. To convert the videos to 30fps, you can have a look at this other conversation: https://github.com/fpv-iplab/rulstm/issues/4#issuecomment-546733817

**Comment by Morethan123** [comment]
Hello, @antoninofurnari 

Under your instruction, i have already convert the EGTEA Gaze+ videos to 30fps which maintain  the length of the video the same.And also decompose the video to frames.

When i am going to extract flow pictures, i use gpu/flow project as you recommended. However, i got cmake error so that c++ source program can't translate to the target program to execute.I also search to fix the error using google and stackoverflow post for a week while the solution doesn't work for me.

I turn to TSN project , and ready to use the author's code to extract optical flow. But his dependcies only require mac or Ubantu system, while my pc use Window 10 as environment. This make me crazy.

Affected by Corona Virus Disease 2019,  now i have no access to the workstation in college.Would you please provide the features(format lmdb) on EGTEA Gaze+.

**Comment by antoninofurnari** [comment]
Hi,
I'm sorry to hear about troubles due to COVID-19 at your university. 

I got the chance to rescue the features extracted from EGTEA Gaze+ used for the experiments. You can download them from here: https://iplab.dmi.unict.it/sharing/rulstm/features/egtea.zip.

Please also see the README for additional information.

**Comment by Morethan123** [comment]
Hi @antoninofurnari 

Thank you for coming to my aid and releasing features on EGTEA Gaze+.

I have downloaded the file(egtea.zip). When i get the processed data, i am going to prepare split files and action annotations. The action annotations (action.csv) contains action id, action label value, verb(index), noun(index).The action.csv i write 106 actions with voun,noun index. I turn to make sequentially training.csv which should contain video id, video name, starting time,ending time,action index,verb index,noun index. 

However,i noticed that if we use  the official splits proposed by the authors of EGTEA Gaze+ as train split and validation split, in their EGTEA Gaze+ dataset raw annotation, some videos in the train_split1(for example) do not have relevant information in the raw annotation files (action_label.csv).  In other word, it means that i can't record these videos in official splits into training.csv because they do not own  the action's starting time and ending time and video id. How do you handle this situations.

By the way, i also want to know which name you choose as the video name.For example, P03-R01-PastaSalad-823352-825175-F019756-F019809  or  P03-R01-PastaSalad-823352-825175?  Would you still keep the relevant annotations and splits used for rulstm experiment.

Sorry to disturb you again. Hope ro receive your reply. Thank you very much!

**Comment by antoninofurnari** [comment]
Hello,

Regarding the official split, if there are videos without annotations (starting and ending times), I have not included those in the csv files. As per the video name, I specified the filename of the whole video, e.g., P05-R01-PastaSalad.

Hope this helps.

Antonino

**Comment by Morethan123** [comment]
Hi, @antoninofurnari 

Under your instruction, I have already reproduce your result on EGTEA Gaze + as you reported in the paper.  

Now, could i ask you another question about your sample strategy? I notice that you set the S_enc to 6 , S_ant to 8 , and alpha to 0.25s . If i want to change these setting, for instance, Set S_enc = 27,  S_ant =8 , alpha to 0.1s , which  keep the total length of the video is also 3.5s. Can my hypothesis  setting finish training procedures within your provided features?

My purpose for this is to explore the LSTM structure of  performance for processing longer sequence. Hope to receive your reply. Thank you!

**Comment by antoninofurnari** [comment]
Hello,

Yes, you should be able to train the model by changing the settings as you suggest, provided that you load the lmdb dataset with all the features (not just the subset used for the default settings).

Best,
Antonino

**Comment by zerodecoder1** [comment]
Hi,

I had a quick question regarding training a model for EGTEA -- can we use the same training script as for epic kitchens but just load in data using the new database? Or what would be the easiest way to train a model using this dataset?

Thank you!

**Comment by antoninofurnari** [comment]
Hello,

Yes, that should be the fastest way to perform training. Since the csvs formatted for RULSTM are shared in https://github.com/fpv-iplab/rulstm/tree/master/RULSTM/data/egtea, that should be straightforward.

However, please consider that the standard evaluation scheme of EGTEA-Gaze+ is a 3-fold cross validation, so you should train three models on the three splits using the following train/test csvs:
  - Split1: `training1.csv` for training and `validation1.csv` for test;
  - Split2: `training2.csv` for training and `validation2.csv` for test;
  - Split3: `training3.csv` for training and `validation3.csv` for test;

You should then report the average of the results that you obtain on the three splits.

Best,
Antonino
